model/src/main/kotlin/DependencyNavigator.kt

```diff
@@ -95,9 +97,69 @@ interface DependencyNavigator {
         matcher: DependencyMatcher = MATCH_ALL
     ): Set<Identifier>
 
+    /**
+     * Return a map with the shortest paths to each dependency in all scopes of the given [project]. The path to a
+     * dependency is defined by the nodes of the dependency tree that need to be passed to get to the dependency. For
+     * direct dependencies the shortest path is empty. The resulting map has scope names as keys; the values are maps
+     * with the shorted paths to all the dependencies contained in that scope.
+     */
+    fun getShortestPaths(project: Project): Map<String, Map<Identifier, List<Identifier>>> =
+        getShortestPathForProject(scopeDependencies(project)) { directDependencies(project, it) }
+
     /**
      * Return the set of [Identifier]s that refer to sub-projects of the given [project].
      */
     fun collectSubProjects(project: Project): SortedSet<Identifier> =
         scopeDependencies(project, matcher = MATCH_SUB_PROJECTS).collectDependencies()
+
+    /**
+     * Determine the map of shortest paths for all the dependencies of a project, given its map of [scopeDependencies]
+     * and a [scopeResolver] function that retrieves the direct dependency nodes of a scope.
+     */
+    private fun getShortestPathForProject(
+        scopeDependencies: Map<String, Set<Identifier>>,
+        scopeResolver: (String) -> Sequence<DependencyNode>
```

Comment:
Is there a case where this is not the `directDependencies()` function? If not I would inline the function to call to simplify the signature.

Reply:
Done.